implementations/rust/ockam/ockam_transport_udp/src/messages/routing_number.rs

```diff
@@ -0,0 +1,97 @@
+use core::fmt::{Debug, Formatter};
+use minicbor::{CborLen, Decode, Encode};
+use rand::random;
+use std::cmp::Ordering;
+
+/// Number of the [`UdpRoutingMessage`]. Each [`UdpRoutingMessage`] is assigned a value, which
+/// helps the receiver to assemble the message. Start with a random value and uses overflowing
+/// arithmetics for increment.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Encode, Decode, CborLen)]
+#[cbor(transparent)]
+pub struct RoutingNumber(#[n(0)] pub u16);
+
+impl core::fmt::Display for RoutingNumber {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        core::fmt::Display::fmt(&self.0, f)
+    }
+}
+
+impl RoutingNumber {
+    pub fn new() -> Self {
+        Self(random())
+    }
+
+    pub fn increment(&mut self) {
```

Comment:
I know that this all works ok in the current implementation but I'm slightly uneasy with having the `increment` logic inside the `RoutingNumber`. I'd be more comfortable with:

- Having `RoutingNumber` being immutable.
- Have the `increment` logic as a function `RoutingNumber -> RoutingNumber` inside the `UdpSenderWorker` since this is the only place where this should be called.